python/tests/sbp/client/test_handler.py

```diff
@@ -105,26 +84,71 @@ def test_handler_callbacks():
   handler.remove_callback(global_counter2)
   handler.remove_callback(msg_type_counter1, 0x55)
   handler.remove_callback(msg_type_counter2, 0x66)
-  handler.call(SBP(0x11, None, None, None, None))
-  handler.call(SBP(0x55, None, None, None, None))
+  handler._call(SBP(0x11, None, None, None, None))
+  handler._call(SBP(0x55, None, None, None, None))
   assert global_counter1.value == 2
   assert global_counter2.value == 2
   assert msg_type_counter1.value == 1
   assert msg_type_counter2.value == 0
 
 def test_multiple_handler_callbacks():
-  handler = Handler(None, None)
+  handler = Handler(())
   msg_type_counter1 = TestCallbackCounter()
   msg_type_counter2 = TestCallbackCounter()
   handler.add_callback(msg_type_counter1, [0x55, 0x66])
   handler.add_callback(msg_type_counter2, [0x11, 0x55])
-  handler.call(SBP(0x11, None, None, None, None))
-  handler.call(SBP(0x55, None, None, None, None))
+  handler._call(SBP(0x11, None, None, None, None), x=1, y=6)
+  handler._call(SBP(0x55, None, None, None, None))
   assert msg_type_counter1.value == 1
   assert msg_type_counter2.value == 2
   handler.remove_callback(msg_type_counter1, [0x55, 0x66])
   handler.remove_callback(msg_type_counter2, [0x11, 0x55])
-  handler.call(SBP(0x11, None, None, None, None))
-  handler.call(SBP(0x55, None, None, None, None))
+  handler._call(SBP(0x11, None, None, None, None), delta=0)
+  handler._call(SBP(0x55, None, None, None, None), something=False)
   assert msg_type_counter1.value == 1
   assert msg_type_counter2.value == 2
+
+def test_child_iter():
+  msgs = ((MsgLog(level=1, text="Hello"), {'stuff':True}),
+          (MsgLog(level=1, text="Hello"), {'delta':0}),
+          (MsgLog(level=1, text="Hello"), {}))
+  handler = Handler(msgs)
+  def latestart():
+    time.sleep(0.1)
+    handler.start()
+  threading.Thread(target=latestart).start()
+  recvd = tuple(handler)
+  assert recvd == msgs
+
+def test_filter():
+  msgs = ((SBP(0x11, None, None, None, None), {}),
+          (SBP(0x33, None, None, None, None), {}),
+          (SBP(0x55, None, None, None, None), {}))
+  handler = Handler(msgs)
+  def latestart():
+    time.sleep(0.1)
+    handler.start()
+  threading.Thread(target=latestart).start()
+  middle = tuple(handler.filter(0x33))
+  assert len(middle) == 1
+  assert middle[0] == msgs[1]
+
+def test_dead_gc():
+  handler = Handler(((SBP(0x11, None, None, None, None), {}),))
+  def latestart():
+    time.sleep(0.1)
+    handler.start()
+  thread = threading.Thread(target=latestart)
+  thread.start()
+  iter(handler) # Dead iterator, should be gc'd
+  xx = iter(handler)
+  thread.join()
+  handler._receive_thread.join()
+  gc.collect()
+  assert [sink() for sink in handler._sinks] == [xx]
+
```

Comment:
This `test_dead_gc` does cover the dead callback exception.  The exception is an internal implementation detail in the removing of the weakref to the gc'd iterator.